File: dotnet/Selenium/Request/ModifyMethodTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.BiDi.Network;

namespace Selenium.Request;

[TestClass]
public class ModifyMethodTest : TestBase
{
    [TestMethod]
    public async Task ModifyHttpMethod()
    {
        await using var intercept = await Bidi.Network.InterceptRequestAsync(async req =>
        {
            await req.ContinueAsync(new ContinueRequestOptions { Method = "HEAD" });
        });

        // var options = new RequestInterceptOptions {
        //               UrlPatterns = new List<UrlPattern> { new UrlPattern("https://www.selenium.dev/") },
        //           };
        // await using var registration = await driver.Network.AddRequestHandlerAsync(
        //     options,
        // .   req =>
        //     {
        //          req.Method = "HEAD";
        //      }
        // );
        
        await Driver.Navigate().GoToUrlAsync("https://selenium.dev");

        Assert.AreEqual("", Driver.FindElement(By.TagName("body")).Text);
    }
}

File: java/selenium/src/test/java/dev/selenium/request/ModifyMethodTest.java
package dev.selenium.request;

import dev.selenium.TestBase;
import java.net.URI;
import java.util.function.Predicate;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.remote.http.HttpMethod;
import org.openqa.selenium.remote.http.HttpRequest;

public class ModifyMethodTest extends TestBase {

  @Test
  public void modifyHttpMethod() {
    Predicate<URI> filter = uri -> uri.getHost().equals("selenium.dev");

    ((FirefoxDriver) driver)
        .network()
        .addRequestHandler(filter, req -> new HttpRequest(HttpMethod.HEAD, req.getUri()));

      // .  RequestInterceptOptions options =
      // RequestInterceptOptionsBuilder.addFilter(Map.of("host", "httpbin.org").build();
    //    driver.network().addRequestHandler(options, req -> req.setMethod("HEAD"));

    driver.get("https://selenium.dev");

    Assertions.assertEquals("", driver.findElement(By.tagName("body")).getText());
  }
}
